src/python/espressomd/reaction_ensemble.pyx

```diff
@@ -357,6 +357,11 @@ cdef class ConstantpHEnsemble(ReactionAlgorithm):
     def __dealloc__(self):
         del(self.constpHptr)
 
+    def add_reaction(self, *args, **kwargs):
+        if(len(kwargs["product_types"])!=2 or len(kwargs["reactant_types"])!=1):
+            raise ValueError("The constant pH method is not implemented for this kind of reactions.")
```

Comment:
Maybe better state that constant pH requires "product_types" == 2 and "reactant_types" == 1.

Reply:
it shall raise an error if(len(kwargs["product_types"])!=2 or len(kwargs["reactant_types"])!=1). 
you could rewrite as:
if(len(kwargs["product_types"])==2 or len(kwargs["reactant_types"])==1):
   pass
else:
   raise...

I do not see why doing it though.